shared/dataflow/codeql/dataflow/DataFlowImpl.qll

```diff
@@ -3981,7 +4003,7 @@ module MakeImpl<DataFlowParameter Lang> {
 
       private predicate relevantState(FlowState state) {
         sourceNode(_, state) or
-        sinkNode(_, state) or
+        sinkNodeWithState(_, state) or
```

Comment:
We can make another predicate to do a best-effort cartesian approximation for the reverse flow exploration case - this doesn't have to be extremely performant as it's just a debugging tool that'll often be restricted to specific sources/sinks anyway.
Add the following just below the `relevantState` predicate:
```
private predicate revSinkNode(NodeEx node, FlowState state) {
  sinkNodeWithState(node, state)
  or
  Config::isSink(node.asNode()) and
  relevantState(state) and
  not fullBarrier(node) and
  not stateBarrier(node, state)
}
```
and use it in the two places below.